modules/src/core/ics02_client/context.rs

```diff
@@ -58,6 +59,19 @@ pub trait ClientReader {
     /// Returns the current height of the local chain.
     fn host_height(&self) -> Height;
 
+    /// Returns the current timestamp of the local chain.
+    fn host_timestamp(&self) -> Timestamp {
+        let latest_consensus_state = self
+            .host_consensus_state(Height::zero())
+            .expect("host must have pending consensus state");
+        latest_consensus_state.timestamp()
```

Comment:
Do we need a `pending_block` field just to mock the host timestamp? Would it work if we just get the latest block and add a fixed offset to the timestamp of the latest block?

Reply:
Thanks for reviewing @Soares! This is an interesting observation. :+1: IMHO it is still helpful to keep a `pending_block`. I think by having a `pending_block`, the mock impl is closer to a real chain context impl and serves as a reference implementation. This makes it easier to convey the expectation that the chain context is in a state where it is ready to execute a pending block and also allows us to use the `ClientReader::host_timestamp()` provided trait method (just as it is expected to be used by a light client impl).

Reply:
I realized that we don't treat the mock context as a ref impl, so your suggestions make sense. 👍 https://github.com/informalsystems/ibc-rs/pull/1771/commits/3b93cb813fa4edb0eeaf7e19dfaa0462a62fc8a8